test/langtools/tools/javac/records/RecordsBinaryCompatibilityTests.java

```diff
@@ -0,0 +1,307 @@
+                    public static int i() { return 0; }
+                }
+                """,
+                """
+                package pkg;
+                public record R(int i) {}
+                """,
+                """
+                package pkg;
+                public class Client {
+                    public static void main(String... args) {
+                        R r = new R(0);
+                        System.out.println(r.i());
+                    }
+                }
+                """,
+                true,
+                "java.lang.IncompatibleClassChangeError"
+        );
+    }
+
+    @Test
+    public void testCompatibilityAfterChangingRecordComponent3(Path base) throws Exception {
+        testCompatibilityAfterModifyingRecord(
+                base,
+                """
+                package pkg;
+                public record R(int i) {
+                }
+                """,
+                """
+                package pkg;
+                public record R(int j) {}
+                """,
+                """
+                package pkg;
+                public class Client {
+                    public static void main(String... args) {
+                        R r = new R(0);
+                        System.out.println(r.i());
+                    }
+                }
+                """,
+                true,
+                "java.lang.NoSuchMethodError"
+        );
+    }
+
+    /* 1- compiles the first version of the record class source code along with the client source code
+     * 2- executes the client class just to make sure that it works
+     * 3- compiles the second version of the record class
+     * 4- executes the client class and makes sure that the VM throws the expected error or not
+     *    depending on the shouldFail argument
+     */
+    private void testCompatibilityAfterModifyingRecord(
+            Path base,
+            String recordCode1,
+            String recordCode2,
+            String clientCode,
+            boolean shouldFail,
+            String expectedError) throws Exception {
+        Path src = base.resolve("src");
+        Path pkg = src.resolve("pkg");
+        Path recordSrc = pkg.resolve("R");
+        Path client = pkg.resolve("Client");
+
+        tb.writeJavaFiles(recordSrc, recordCode1);
+        tb.writeJavaFiles(client, clientCode);
+
+        Path out = base.resolve("out");
+        Files.createDirectories(out);
+
+        new JavacTask(tb)
+                .outdir(out)
+                .files(findJavaFiles(pkg))
+                .run();
+
+        // let's execute to check that it's working
+        String output = new JavaTask(tb)
+                .classpath(out.toString())
+                .classArgs("pkg.Client")
+                .run()
+                .writeAll()
+                .getOutput(Task.OutputKind.STDOUT);
+
+        // let's first check that it runs wo issues
+        if (!output.contains("0")) {
```

Comment:
This is a very weak test that may easily fail for the wrong reasons.  Can you make the reference value more distinctive, like 12345, or "Hello World" that could never appear accidentally